### PR TITLE
Added tests, checking high voltage status, reading other pressure units, etc

### PR DESCRIPTION
**Pull request by xkstein**
Hi, I've added a few things to your code,

- Tests
- `getHighVoltageStatus()`
- Timeout setting
- Reading pressure units (rather than throwing an exception when they aren't mBar)
- Raising `ConnectionError` with a message rather than just generic runtime error

I also ran it through pylint (hence the changes from `repl == False` to `repl is False`).

I did sort of make a lot of changes though (some of which are just stylistic), so I'd understand if you don't want to merge this. Just figured I'd offer!

### CODE REVIEW

**Comment by tspspi** [comment]
Wow, just skimmed the code (will look into it later) and it looks really great - but I've a small problem with the unit modification on the get_pressure method to merge since this modifies the return value from a single float (in mbar) to a tuple containing value & unit which basically is a breaking API change that would require modification of all code depending on the implementation. Would it be possible to do this either in a new method / a parameter that switches output format defaulting to mbar (maybe converting from torr in case they are selected) or something similar that does not break the API?

File: src/gammaionctl/gammaionctl.py [hunk 1]
```diff
@@ -79,78 +86,92 @@ def identify(self):
...
+        response = self.getPressureWithUnits(pumpIndex)
+        if not isinstance(response, tuple):
+            return response
+        pressure, units = response
+
+        if units.lower() != require_units.lower():
+            if self.verbose:
+                print(f"Gamma QPC not set to {require_units}")
+            return False
+        return pressure
```
[the remark this answers is not in the document]
Comment (tspspi) [reply src/gammaionctl/gammaionctl.py:145]: @xkstein thanks for adressing this - yeah this looks great after skimming fast - I'm going to look over in detail in our local evening & then try to merge the PR into the project (Never done this on GitHub before). Many thanks for adressing the compatibility part. (When I've initially hacked the library I dropped the support for torr & didn't implement tests, etc. out of timing constraints and have been pretty new to Python - in fact this has been one of my first Python projects to be honest)

**Review by tspspi — APPROVED** [review]

File: src/gammaionctl/gammaionctl.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (xkstein) [reply src/gammaionctl/gammaionctl.py:145]: @tspspi Thanks for merging, also my first PR. Thanks for making this utility, and working with me on this!

**Comment by tspspi** [comment]
@xkstein  By the way I always forget to think about this but I think we should mention you in the copyright line in the LICENSE file ...

**Comment by xkstein** [comment]
@tspspi Oh, I also hadn't thought of that, I don't know much about what people normally do for that. It's still mostly your code so we don't need to bother

**Pull request merged**